trieste/acquisition/function/active_learning.py

```diff
@@ -0,0 +1,234 @@
+
+    It turns out that probabilistic models can be used as classifiers for identifying where excursion probability is larger than 1/2 and this idea is used to build many sequential sampling strategies. We follow :cite:`bect2012sequential` and use a formulation which provides a common expression for these two criteria:
+
+    .. math:: E[\max(0, (\alpha s(x))^\delta - |T - m(x)|^\delta)]$
+
+    Here :math:`m(x)` and :math:`s(x)` are the mean and standard deviation of the predictive posterior of a probabilistic model. Bichon criterion is obtained when :math:`\delta = 1` while ranjan criterion is obtained when :math:`\delta = 2`. :math:`\alpha>0` is another parameter that acts as a percentage of standard deviation of the posterior around the current boundary estimate where we want to sample. The goal is to sample a point with a mean close to the threshold :math:`T` and a high variance, so that the positive difference in the equation above is as large as possible.
+    
+    Only batches of size 1 are allowed.
+
+    :param model: The probabilistic model of the objective function.
+    :param threshold: The failure or feasibility threshold.
+    :param alpha: The parameter which determines the neihbourhood around the estimated contour
+        line as a percentage of the posterior variance in which to allocate new points.
+    :param delta: The parameter identifying which criterion is returned, Bichon for value of 1
+        and Ranjan for value of 2.
+    """
+
+    @tf.function
+    def acquisition(x: TensorType) -> TensorType:
+
+        tf.debugging.assert_shapes(
+            [(x, [..., 1, None])],
+            message="This acquisition function only supports batch sizes of one.",
+        )
+
+        mean, variance = model.predict(tf.squeeze(x, -2))
+        stdev = tf.sqrt(variance)
+        t = (mean - threshold) / stdev
+        t_plus = t + alpha
+        t_minus = t - alpha
+        normal = tfp.distributions.Normal(tf.cast(0, x.dtype), tf.cast(1, x.dtype))
+
+        if delta == 1:
+            G = (
+                alpha * (normal.cdf(t_plus) - normal.cdf(t_minus))
+                - t * (2 * normal.cdf(t) - normal.cdf(t_plus) - normal.cdf(t_minus))
+                - (2 * normal.prob(t) - normal.prob(t_plus) - normal.prob(t_minus))
+            )
+            criterion = G * stdev
+        elif delta == 2:
+            G = (
+                (alpha ** 2 - 1 - t ** 2) * (normal.cdf(t_plus) - normal.cdf(t_minus))
+                - 2 * t * (normal.prob(t_plus) - normal.prob(t_minus))
+                + t_plus * normal.prob(t_plus)
+                - t_minus * normal.prob(t_minus)
+            )
+            criterion = G * variance
+
+        return criterion
```

Comment:
if I understand the papers correctly this quantity is meant to be maximized, so we should return the negative here? perhaps also name it `NegativeExpectedFeasibility` consistent with LCB?

Reply:
if this criterion is maximised, then we should not do anything? LCB should be minimised, which is why we add a negative.

Reply:
right, I managed to miss the minus part here in the `../optimizer.py`:
```
def _objective() -> TensorType:
            return -target_func(variable[:, None, :])  # [1]
```